src/psyclone/tests/domain/lfric/transformations/raise_psyir_2_lfric_kern_trans_test.py

```diff
@@ -325,7 +325,7 @@ def test_apply_ok(fortran_reader):
         "TYPE, PUBLIC, EXTENDS(kernel_type) :: testkern_type\n"
         "  type(ARG_TYPE) :: META_ARGS(1) = (/ &\n"
         "    arg_type(gh_field, gh_real, gh_inc, w1)/)\n"
-        "  INTEGER :: OPERATES_ON = cell_column\n"
+        "  INTEGER :: OPERATES_ON = owned_cell_column\n"
```

Comment:
This OPERATES_ON change is repeated in multiple places, and it make me a bit uncomfortable because it breaks the separations of concerns. Now scientists (who write algorithm and kernels) need to be aware about distributed memory, domain partitions and halos.

I know this is exactly what this feature needs, but I am wondering if we should limit this to the "halo_cell_columns" as a specific optimisation, but revert/delete the "owned_cell_column" and "owned_and_halos_cell_column"

The first because I think most people should just use "cell_column" and not even consider that there is a halo, or what "owned" means.

The second because there shouldn't be a use case for them from the DSL point of view. Is the rules that decide if we iterate over owned of everything.

What do you think?

Reply:
Ok, there seems to be agreement in removing the "owned_cell_column" and leave it as "cell_column" (this was my main concern, to leave the simple-case simple).

Also, there are use cases for both "HALO_CELL_COLUMN" and "OWNED_AND_HALO_CELL_COLUMN" to provide specialised optimisations which I agree feel more natural as specific OPERATES_ON options rather than IR transformations.

So for this PR I will just request to revert the "OWNED_CELL_COLUMN".

(And not for this issue-PR, but I wonder if this is all necessary because the extended mesh concept is not part of the DSL/PSyKAl API. Maybe if psyclone knew about them it could do some of these optimisations instead of being kernels. But this would be a much bigger design/implementation effort and not sure if its worth it)

Reply:
Thanks Tom, it is really helpful to get this all written down. I agree that a transformation in order to generate scientifically-correct code is a Bad Idea - a kernel's metadata should completely capture everything that is needed to correctly run that kernel. It's unfortunate that there's just one kernel that requires `OWNED_AND_HALO_CELL_COLUMN` (although, now that I write that, it's only a relatively small part of this PR). Just in case, would it be possible to have this as two separate kernels, one with CELL_COLUMN and one with HALO_CELL_COLUMN or would that just cause other problems (e.g. if someone used the redundant-computation transformation on the former)?

Reply:
I think that's right, it would be possible to have two separate kernels -- the only downside is that they would be duplicates save for the `operates_on` metadata

Reply:
That doesn't sound too bad - they could both call a separate routine with the actual implementation. (Although it would currently inhibit our ability to offload the kernel to GPU but a solution to that is in the works.) What do you think @sergisiso ?

Reply:
I slightly prefer having the `OWNED_AND_HALO_CELL_COLUMN` than requiring 2 duplicate kernels with different `operates_on`, even if they are wrappers to the same functionality.